app/screens/RemeasurementFlow/RemeasurementReview.tsx

```diff
@@ -0,0 +1,543 @@
+  updatePlantLocationHistory,
+  updatePlantLocationHistoryEventDate,
+  updatePlantLocationHistoryStatus,
+} from '../../repositories/plantLocationHistory';
+import {getUserInformation} from '../../repositories/user';
+import {Colors, Typography} from '../../styles';
+import {DBHInMeter, nonISUCountries} from '../../utils/constants';
+import {
+  EDITING,
+  FIX_NEEDED,
+  INCOMPLETE,
+  INCOMPLETE_SAMPLE_TREE,
+  PENDING_DATA_UPLOAD,
+  SYNCED,
+} from '../../utils/inventoryConstants';
+import {getConvertedDiameter, getConvertedHeight} from '../../utils/measurements';
+import {measurementValidation} from '../../utils/validations/measurements';
+const {protocol, cdnUrl} = APIConfig;
+
+type Props = {};
+
+export default function RemeasurementReview({}: Props) {
+  const [isOpenModal, setIsOpenModal] = useState(false);
+  const [diameter, setDiameter] = useState('');
+  const [editableDiameter, setEditableDiameter] = useState('');
+  const [height, setHeight] = useState('');
+  const [editableHeight, setEditableHeight] = useState('');
+  const [editEnabledFor, setEditEnabledFor] = useState('');
+  const [isNonISUCountry, setIsNonISUCountry] = useState(false);
+  const [showIncorrectRatioAlert, setShowIncorrectRatioAlert] = useState(false);
+  const [imageUrl, setImageUrl] = useState('');
+  const [plantLocationHistory, setPlantLocationHistory] = useState({});
+  const [HID, setHID] = useState('');
+  const [showDeleteAlert, setShowDeleteAlert] = useState(false);
+
+  const [inputErrorMessage, setInputErrorMessage] = useState<string>(
+    i18next.t('label.tree_inventory_input_error_message'),
+  );
+  const [showInputError, setShowInputError] = useState<boolean>(false);
+  const [dataStatus, setDataStatus] = useState<string>('');
+  const [oldDataStatus, setOldDataStatus] = useState('');
+
+  const [diameterLabel, setDiameterLabel] = useState<string>(
+    i18next.t('label.measurement_basal_diameter'),
+  );
+
+  const [isEditable, setIsEditable] = useState<boolean>(false);
+
+  // statues under which editing of data is possible
+  const editableStatusCondition = [
+    INCOMPLETE,
+    INCOMPLETE_SAMPLE_TREE,
+    FIX_NEEDED,
+    EDITING,
+    PENDING_DATA_UPLOAD,
+  ];
+
+  const navigation = useNavigation();
+
+  const {
+    state: {selectedRemeasurementId},
+    dispatch,
+  } = useContext(InventoryContext);
+
+  useEffect(() => {
+    // used to check if the user is in non-ISU country
+    getUserInformation().then(user => {
+      setIsNonISUCountry(nonISUCountries.includes(user?.country || ''));
+    });
+
+    console.log(selectedRemeasurementId, 'selectedRemeasurementId');
+
+    // used to get the data for the selected remeasurement
+    getPlantLocationHistoryById(selectedRemeasurementId).then((plantLocationHistory: any) => {
+      getInventoryByLocationId({locationId: plantLocationHistory?.parentId})
+        .then(inventory => {
+          console.log(plantLocationHistory, ' plantLocationHistory.samplePlantLocationIndex');
```

Comment:
The app got stuck here after trying to remeasure a sample tree of a multiple-tree registration with some endless spinner at the following log output:

```
{"isRemeasurement": true, "isTreeAlive": true} route?.params
 LOG  multi REMEASUREMENT false true =====
 LOG  1
 LOG  [{"key":"appVersion","value":"1.0.9","accessType":"app"}] transformedData true
 LOG  2
 LOG  [{"accessType": "app", "key": "appVersion", "value": "1.0.9"}] additionalData
 LOG  3
 LOG  true isRemeasurement
 LOG  IldhhrNL982uTuA-HnVdM selectedRemeasurementId
 LOG  undefined  plantLocationHistory.samplePlantLocationIndex
 LOG  [TypeError: undefined is not an object (evaluating 'plantLocationHistory.samplePlantLocationIndex')]

```
![Simulator Screen Shot - iPhone 12 - 2023-01-21 at 13 19 24](https://user-images.githubusercontent.com/1532418/213866589-d857d2be-52d0-4263-972e-4838f715eafc.png)

Reply:
This happens doing a remeasurement after opening a multiple tree registration from the tree inventory list of uploaded items. It does not happen when starting the remeasurement if triggered by selected the polygon on the map and directly hit the Remeasure button of a sample tree in the preview box on the hud display on the map.

![Simulator Screen Shot - iPhone 12 - 2023-01-21 at 14 36 26](https://user-images.githubusercontent.com/1532418/213869354-e735f958-7b43-42a9-99cf-c6dc6ed81baa.png)![Simulator Screen Shot - iPhone 12 - 2023-01-21 at 14 36 37](https://user-images.githubusercontent.com/1532418/213869355-27785fa8-7429-468c-8c33-a8cff9478900.png)

Reply:
Maybe related issue is https://github.com/Plant-for-the-Planet-org/treemapper/issues/658.